### DISCUSSION TITLE
Data types & sorting

### DISCUSSION THREAD

**Discussion by fadinasr** [prisma/prisma#21685] [Q&A] [answered]
### Question

We have a table that links a value (JSONB column) to a `fieldID` coming from another table. The JSONB column can have multiple types (string, numeric, date, etc ...) and we depend on the field type to deal with that. We are currently facing a problem sorting the values we receive from this column. For example: 

Expected behaviour: 1, 2, 3, 11, 22, 3, 33
Current behaviour: 1, 11, 2, 22, 3, 33

We thought of a workaround of saving the values of the fields per type in different typed columns other than the JSONB field so we could deal with the types correctly. However, this solution isn't ideal as it might introduce many bugs and data-related issues.

P.S. We are trying to use Prisma queries and get rid of all the raw queries that has to deal with this functionality and similar functionalities.

Could you provide any recommendations?

### How to reproduce (optional)

1. Create two tables, one for a field that has a name and a type and another one for the value (`JSONB`) that has a field relationship
2. Query the values from the values table with order by
3. You'll receive similar results to the one in the description 

### Expected behavior (optional)

Being able to sort numeric fields correctly (numeric sorting not character-based sorting)

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
model DealFieldValue {
  id    String @id @unique @default(uuid()) @db.Uuid
  value Json   @default(dbgenerated())

  createdAt DateTime @default(now()) @db.Timestamptz
  updatedAt DateTime @default(now()) @updatedAt @db.Timestamptz

  deal   Deal?   @relation(fields: [dealId], references: [id], onDelete: Cascade)
  dealId String? @db.Uuid

  field   Field  @relation(fields: [fieldId], references: [id], onDelete: Cascade)
  fieldId String @db.Uuid

  @@unique([fieldId, dealId])
  @@index([fieldId])
  @@index([dealId])
}

model Field {
  id                 String   @id @unique @default(uuid()) @db.Uuid
  name               String
  label              String
  dataType           String
  description        String?
  order              Int
  visible            Boolean
  required           Boolean  @default(false)

  createdAt DateTime @default(now()) @db.Timestamptz
  updatedAt DateTime @default(now()) @updatedAt @db.Timestamptz

  dealFieldValue    DealFieldValue[]

  @@map("crm_field")
}

```

- OS: Any OS
- Database: PostgreSQL
- Node.js version: v16.16.X
- Prisma: v4.15.0

**Comment by ludralph** [comment, marked as answer]
Hi @fadinasr 👋 

Thank you for raising this question.

Prisma currently does not support ordering by JSONB columns. This is confirmed by this open GitHub [issue](https://github.com/prisma/prisma/issues/9489). 

The user in the issue describes a similar problem where they are unable to sort by values inside of the Postgres JSONB column using Prisma and have to resort to raw SQL queries using [$queryRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw). 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
